lib/foreman/renderer/scope/report.rb

```diff
@@ -2,6 +2,46 @@ module Foreman
   module Renderer
     module Scope
       class Report < Foreman::Renderer::Scope::Base
+        def initialize(**args)
+          super
+          @report_data = []
+          @report_headers = []
+        end
+
+        def report_render(format: :csv)
+          case format
+          when :csv
+            report_render_csv
+          when :yaml
+            report_render_yaml
+          end
+        end
+
+        def report_row(row_data)
+          @report_headers = row_data.keys.map(&:to_s) if @report_headers.empty?
+          @report_data << row_data.values.map do |cell|
+            cell.is_a?(Enumerable) ? cell.map(&:to_s) : cell.to_s
```

Comment:
If we call to_s here, would json formatter (in future) be able to display numbers as integers? Perhaps that should be call of each formatter.